google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/stackdriver/facet/StackdriverPanel.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.stackdriver.facet;
+
+import com.google.cloud.tools.intellij.util.GctBundle;
+
+import com.intellij.facet.ui.FacetEditorContext;
+import com.intellij.facet.ui.FacetEditorTab;
+import com.intellij.openapi.options.ConfigurationException;
+
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.JCheckBox;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+
+/**
+ * Stackdriver facet's user interface.
+ */
+public class StackdriverPanel extends FacetEditorTab {
+
+  private JCheckBox generateSourceContext;
+  private JPanel stackdriverPanel;
```

Comment:
if this is the enclosing panel, I'd be tempted to put it at the top before I declare the elements that it's composed of.